src/Text/Markdown/SlamDown/Parser/Inline.purs

```diff
@@ -1,336 +1,436 @@
 module Text.Markdown.SlamDown.Parser.Inline
   ( parseInlines
-  , validateTextOfType
   , validateFormField
   , validateInline
-  )
-  where
+  , parseTextBox
+  ) where
 
 import Prelude
 
 import Control.Alt ((<|>))
 import Control.Apply ((<*), (*>))
-import Control.Lazy (fix)
+import Control.Lazy as Lazy
 
+import Data.Const (Const(..))
 import Data.Either (Either(..))
 import Data.Foldable (elem)
-import Data.List (List(..), take, many, some, singleton, fromList, length)
-import Data.Maybe (Maybe(..), maybe)
+import Data.Functor ((<$))
+import Data.Functor.Compose (Compose(..))
+import Data.HugeNum as HN
+import Data.List as L
+import Data.Maybe as M
 import Data.Traversable (traverse)
 import Data.Tuple (Tuple(..), fst, snd)
-import Data.Validation (V(), invalid, runV)
-import qualified Data.String as S
+import Data.Validation as V
+import Data.String as S
 
-import Text.Parsing.Parser (Parser(), ParseError(..), runParser)
-import Text.Parsing.Parser.Combinators (try, option, optional, optionMaybe, manyTill, many1Till, sepBy, (<?>))
-import Text.Parsing.Parser.String (eof, satisfy, string, anyChar, noneOf)
+import Text.Parsing.Parser as P
+import Text.Parsing.Parser.Combinators as PC
+import Text.Parsing.Parser.String as PS
 
-import Text.Markdown.SlamDown.Syntax
-import Text.Markdown.SlamDown.Parser.Utils
+import Text.Markdown.SlamDown.Syntax as SD
+import Text.Markdown.SlamDown.Parser.Utils as PU
 
-foreign import error :: forall a. String -> a
+foreign import error ∷ ∀ a. String → a
 
-parseInlines :: forall a. (Value a) => List String -> List (Inline a)
-parseInlines s = consolidate $ throwOnError $ runParser (S.joinWith "\n" $ fromList s) inlines
+-- TODO: why is this throwing an error in JavaScript?
```

Comment:
Thanks!

This is referring to the FFI is being used to implement `throwOnError`, which is an unsafe `fromRight`-style function. I *think* that we have the invariant that this parser never fails, so the exception should be benign/unobservable, so there's probably nothing to worry about.

I just found the code a bit strange, before I had realized that the parsers always succeed. I can remove the comment if you're satisfied on that count.